### PR TITLE
Make line and paragraph spacing configurable

### PR DESCRIPTION
**Pull request by Tom94**
Also renames Paragraph to TextFlowContainer to better reflect
its nature of containing potentially multiple paragraphs.

### CODE REVIEW

File: osu.Framework/Graphics/Sprites/Paragraph.cs [hunk 1]
```diff
@@ -12,8 +12,9 @@ namespace osu.Framework.Graphics.Sprites
 {
     /// <summary>
     /// A drawable text object that supports word-wrapping amongst other paragraph-specific formatting.
+    /// A new paragraph is initiated by a linebreak character (i.e. '\n') in text.
     /// </summary>
-    public class Paragraph : FillFlowContainer
```
Comment (peppy) [inline osu.Framework/Graphics/Sprites/Paragraph.cs]: need to rename file

**Comment by default0** [comment]
Maybe eventually restructure the TextFlowContainer to contain Paragraph-instances (which can allow for internal newlines that have 0 spacing, similar to words behaviour on Shift-Enter) and have some helper methods to make working with it easier.
One side effect of this would be that Paragraphs in the TextFlowContainer could have LineSpacing independent of each other.
Another imho big plus of this would be that the behaviour of \n is easier to understand if you can define it separately for TextFlowContainer and Paragraph.
While using, I was quite confused that a class called "Paragraph" interpreted \n as "start a new paragraph", even though all I asked for was one Paragraph (ie: I only made one instance of the Paragraph class).

**Comment by Tom94** [comment]
I think it'd be unnecessarily annoying to make multi-paragraph text if every paragraph needs to be constructed individually. I'm okay with treating '\n' as "new paragraph" since popular document editors like Word also generate new paragraphs on linebreak. If it is desired to have different '\n' spacing at different points in text (i.e. to simulate non-spaced linebreaks within individual paragraphs), then you can always take multiple `TextFlowContainer` and stack them on top of each other via another `FlowContainer`.

**Comment by Tom94** [comment]
Hold off with merging for now, we're still making small changes.

**Comment by Tom94** [comment]
Okay; all good for merge. :)

**Comment by default0** [comment]
Now, TextFlowContainer is great again.

File: osu.Framework/Graphics/Containers/TextFlowContainer.cs [hunk 2]
```diff
@@ -144,18 +151,23 @@ private float getLineHeight(Drawable d)
...
 
-        private IEnumerable<SpriteText> addLine(TextLine line)
+        private IEnumerable<SpriteText> addLine(TextLine line, bool newLineIsParagraph)
         {
             bool first = true;
             var sprites = new List<SpriteText>();
 
+            // !newLineIsParagraph effectively means that we want to add just *one* paragraph, which means we need to make sure that any previous paragraphs
+            // are terminated. Thus, we add a NewLineContainer that indicates the end of the paragraph before adding our current paragraph.
+            if(!newLineIsParagraph)
```
Comment (smoogipooo) [inline osu.Framework/Graphics/Containers/TextFlowContainer.cs]: Add space between if and parens.

**Comment by smoogipooo** [comment]
Looks good ;D

**Pull request merged**